src/CTA.Rules.Actions/ObjectCreationExpressionActions.cs

```diff
@@ -34,5 +36,71 @@ ExpressionSyntax action(SyntaxGenerator syntaxGenerator, ObjectCreationExpressio
+        {
+            Func<SyntaxGenerator, ObjectCreationExpressionSyntax, ExpressionSyntax> action = (SyntaxGenerator syntaxGenerator, ObjectCreationExpressionSyntax node) =>
+            {
+                if(node.Initializer != null && node.Initializer.Expressions.Count > 0)
+                {
+                    var memberList = node.Initializer.Expressions.Where(n => n.Kind() == SyntaxKind.SimpleAssignmentExpression);
+                    if(memberList.Count() > 0)
+                    {
+                        var assignMemberList = memberList.Select(n => (AssignmentExpressionSyntax)n);
+                        var member = assignMemberList.Where(n => n.Left.ToFullString().Contains(oldMember)).FirstOrDefault();
+                        if (member != null)
+                        {
+                            var newExpression = SyntaxFactory.ParseExpression(newMember + "=" + member.Right);
+                            var newNode = node.Initializer.Expressions.Replace(member, newExpression);
+                            node = node.WithInitializer(node.Initializer.WithExpressions(newNode)).NormalizeWhitespace();
+                        }
+                        else
+                        {
+                            var newExpression = SyntaxFactory.ParseExpression(newMember + "=" + newValueIfAdding);
+                            var newNode = node.Initializer.Expressions.Add(newExpression);
+                            node = node.WithInitializer(node.Initializer.WithExpressions(newNode)).NormalizeWhitespace();
+                        }
+                    }
+                }
+                return node;
+            };
+            return action;
+        }
+
+        public Func<SyntaxGenerator, ObjectCreationExpressionSyntax, ExpressionSyntax> GetReplaceObjectPropertyValueAction(string oldMember, string newMember)
+        {
+            Func<SyntaxGenerator, ObjectCreationExpressionSyntax, ExpressionSyntax> action = (SyntaxGenerator syntaxGenerator, ObjectCreationExpressionSyntax node) =>
+            {
+                if (node.Initializer != null && node.Initializer.Expressions.Count > 0)
+                {
+                    var memberList = node.Initializer.Expressions.Where(n => n.Kind() == SyntaxKind.SimpleAssignmentExpression);
+                    if (memberList.Count() > 0)
+                    {
+                        var assignMemberList = memberList.Select(n => (AssignmentExpressionSyntax)n);
+                        var member = assignMemberList.Where(n => n.Right.ToFullString().Contains(oldMember)).FirstOrDefault();
+                        if (member != null)
+                        {
+                            var newExpression = SyntaxFactory.ParseExpression(member.Left + "=" + member.Right.ToString().Replace(oldMember,newMember));
```

Comment:
What do you think about having a helper class to help build certain types of nodes more easily? For example, here (and on lines 52 and 58 above) we're creating assignment expressions using an undefined pattern which could be substituted for something like `NodeFactory.CreateAssignmentExpression(leftOperand, rightOperand)`. 

I think making it easier to construct various node types would help us and the community w/ maintainability and testability, but I also don't think this is an urgent need. If we think there's value there, we can file an issue to add it later.

Reply:
Good idea, Let's add a new issue for it, I don't want to significantly increase the amount of code to be re-reviewed for this CR and this idea can easily find other functions to refactor out of the other actions classes as well to include in this new helper class.